src/components/CreateAccount.js

```diff
@@ -96,24 +96,36 @@ class CreateAccount extends Component {
     }
 
     componentDidUpdate() {
-        const { step, query } = this.state;
+        const { step } = this.state;
         if (step === 'created') {
             if (this.isWhistle()) {
                 window.postMessage('whistle_signup_complete');
             } else {
-                const urlParameters =
-                    query &&
-                    Object.keys(query)
-                        .map(param => `${param}=${query[param]}`)
-                        .join('&');
+                const redirectUri = this.getRedirect();
                 setTimeout(() => {
-                    window.location.href = `${
-                        window.config.DEFAULT_REDIRECT_URI
-                    }?${urlParameters}`;
+                    window.location.href = redirectUri;
                 }, 5000);
             }
         }
     }
+    getRedirect = () => {
+        const { username } = this.state;
+        const { location: { query } } = this.props;
+
+        const usernamePattern = '{{username}}';
+
+        let redirectUri = window.config.DEFAULT_REDIRECT_URI;
+
+        redirectUri = redirectUri.replace(usernamePattern, username);
+
+        const urlParameters =
```

Comment:
This does not follow the spec, we want the query dict keys only to replace what's in the pattern/template.

E.g. `query = {foo: 'bar', baz: 'ignored'}` and `redirect_uri = 'http://foo.com/?bar={{foo}}` should result in `http://foo.com/?bar=bar`

Reply:
Can we end up with a case like:

`query = {foo: 'bar', baz: 'ignored'}`
`redirect_uri = http://foo.com/?bar={{foo}}?fizz={{stuff}}`

In this case, we have nothing to replace the token `{{stuff}}` with. Do we just ignore the token?

Reply:
Good question, I think when `stuff` is unset we should just substitute it for an empty string, so `http://foo.com/?bar=bar&fizz=` in your example above